common/djangoapps/third_party_auth/api/permissions.py

```diff
@@ -2,32 +2,91 @@
 Third party auth API related permissions
 """
 
+import logging
 
-from rest_framework import permissions
-
+from edx_django_utils.monitoring import set_custom_metric
+from edx_rest_framework_extensions.auth.jwt.decoder import decode_jwt_filters
+from edx_rest_framework_extensions.permissions import (
+    IsSuperuser,
+    JwtHasScope,
+    JwtRestrictedApplication,
+    NotJwtRestrictedApplication
+)
+from rest_condition import C
+from rest_framework.permissions import BasePermission, IsAuthenticated
 from third_party_auth.models import ProviderApiPermissions
 
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+
+log = logging.getLogger(__name__)
+
 
-class ThirdPartyAuthProviderApiPermission(permissions.BasePermission):
+class ThirdPartyAuthProviderApiPermission(BasePermission):
     """
     Allow someone to access the view if they have valid OAuth client credential.
-    """
-    def __init__(self, provider_id):
-        """ Initialize the class with a provider_id """
-        self.provider_id = provider_id
 
+    Deprecated: Only works for DOP oauth applications. To be removed as part of DOPrecation.
+
+    """
     def has_permission(self, request, view):
         """
         Check if the OAuth client associated with auth token in current request has permission to access
         the information for provider
         """
-        if not request.auth or not self.provider_id:
+        provider_id = view.kwargs.get('provider_id')
+        if not request.auth or not provider_id:
             # doesn't have access token or no provider_id specified
             return False
 
         try:
-            ProviderApiPermissions.objects.get(client__pk=request.auth.client_id, provider_id=self.provider_id)
+            ProviderApiPermissions.objects.get(client__pk=request.auth.client_id, provider_id=provider_id)
         except ProviderApiPermissions.DoesNotExist:
             return False
 
+        set_custom_metric('deprecated_ThirdPartyAuthProviderApiPermission', True)
         return True
+
+
+class JwtHasTpaProviderFilterForRequestedProvider(BasePermission):
+    """
+    The JWT used to authenticate contains the appropriate tpa_provider
+    filter for the requested provider.
+    """
```

Comment:
Add to comment that this is only used on one endpoint and only used in Edge, so not wanting to pollute the library.